test/e2e/storage/persistent_volumes-volumemode.go

```diff
@@ -0,0 +1,480 @@
+	pvNamePrefix  = "pv"
+)
+
+func generateConfigsForStaticProvisionPVTest(scName string, volBindMode storagev1.VolumeBindingMode,
+	volMode v1.PersistentVolumeMode, pvSource v1.PersistentVolumeSource) (*storagev1.StorageClass,
+	framework.PersistentVolumeConfig, framework.PersistentVolumeClaimConfig) {
+	// StorageClass
+	scConfig := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: scName,
+		},
+		Provisioner:       noProvisioner,
+		VolumeBindingMode: &volBindMode,
+	}
+	// PV
+	pvConfig := framework.PersistentVolumeConfig{
+		PVSource:         pvSource,
+		NamePrefix:       pvNamePrefix,
+		StorageClassName: scName,
+		VolumeMode:       &volMode,
+	}
+	// PVC
+	pvcConfig := framework.PersistentVolumeClaimConfig{
+		AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+		StorageClassName: &scName,
+		VolumeMode:       &volMode,
+	}
+
+	return scConfig, pvConfig, pvcConfig
+}
+
+func createPVTestResource(cs clientset.Interface, ns string,
+	scConfig *storagev1.StorageClass, pvConfig framework.PersistentVolumeConfig,
+	pvcConfig framework.PersistentVolumeClaimConfig) (*storagev1.StorageClass, *v1.Pod, *v1.PersistentVolume, *v1.PersistentVolumeClaim) {
+
+	By("Creating sc")
+	sc, err := cs.StorageV1().StorageClasses().Create(scConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating pv and pvc")
+	pv, pvc, err := framework.CreatePVPVC(cs, pvConfig, pvcConfig, ns, false)
+	framework.ExpectNoError(err)
+	framework.ExpectNoError(framework.WaitOnPVandPVC(cs, ns, pv, pvc))
+
+	By("Creating a pod")
+	// TODO(mkimuram): Need to set anti-affinity with storage server pod.
+	// Otherwise, storage server pod can also be affected on destructive tests.
+	pod, err := framework.CreateSecPod(cs, ns, []*v1.PersistentVolumeClaim{pvc}, false, "", false, false, framework.SelinuxLabel, nil, framework.PodStartTimeout)
+	Expect(err).NotTo(HaveOccurred())
+
+	return sc, pod, pv, pvc
+}
+
+func createPVTestResourceWithFailure(cs clientset.Interface, ns string,
+	scConfig *storagev1.StorageClass, pvConfig framework.PersistentVolumeConfig,
+	pvcConfig framework.PersistentVolumeClaimConfig) (*storagev1.StorageClass, *v1.Pod, *v1.PersistentVolume, *v1.PersistentVolumeClaim) {
+
+	By("Creating sc")
+	sc, err := cs.StorageV1().StorageClasses().Create(scConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating pv and pvc")
+	pv, pvc, err := framework.CreatePVPVC(cs, pvConfig, pvcConfig, ns, false)
+	framework.ExpectNoError(err)
+	framework.ExpectNoError(framework.WaitOnPVandPVC(cs, ns, pv, pvc))
+
+	By("Creating a pod")
+	pod, err := framework.CreateSecPod(cs, ns, []*v1.PersistentVolumeClaim{pvc}, false, "", false, false, framework.SelinuxLabel, nil, framework.PodStartTimeout)
+	Expect(err).To(HaveOccurred())
+
+	By("Checking events for pod")
+	eventList, err := cs.CoreV1().Events(ns).List(metav1.ListOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	for _, event := range eventList.Items {
+		framework.Logf("%v", event.Message)
```

Comment:
This just prints all events. Do you expect any specific event to appear? You should check it's there. Failed tests already print all events to logs, is it enough?